lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -106,6 +106,10 @@ def set_request_tags!(request_span, env, status, headers, response, original_env
           # is just the relative path without query strings.
           url = env['REQUEST_URI'] || original_env['PATH_INFO']
           request_id = get_request_id(headers, env)
+          etag = env['ETAG'] || original_env['ETAG']
+          cache_control = env['CACHE_CONTROL'] || original_env['CACHE_CONTROL']
+          if_modified_since = env['HTTP_IF_MODIFIED_SINCE'] || original_env['HTTP_IF_MODIFIED_SINCE']
+          if_none_match = env['HTTP_IF_NONE_MATCH'] || original_env['HTTP_IF_NONE_MATCH']
```

Comment:
`original_env` is used to to access env entries that could otherwise be changed by other middlewares used by the application. I think in this case checking only `env` should be enough.

Reply:
@dasch The difference between `original_env` and `env` has to do with how `env` gets mutated by Rack applications.

 - `original_env` is a copy of `env` as it was passed into the tracer.
 - `env` is the usual Rack `env`, that gets passed through the Rack application. The Rack application can and will mutate keys/values.

The main motivation here was because Rails mutates `PATH_INFO` and messes with other routing info in a destructive way. Back when we were trying to get error traces to produce correct URLs, (I think) we introduced this to preserve the `env` before Rails messed with the routing info.